mesonbuild/compilers/mixins/clike.py

```diff
@@ -1269,7 +1269,7 @@ def _has_multi_arguments(self, args: T.List[str], env: 'Environment', code: str)
             # some compilers, e.g. GCC, don't warn for unsupported warning-disable
             # flags, so when we are testing a flag like "-Wno-forgotten-towel", also
             # check the equivalent enable flag too "-Wforgotten-towel"
-            if arg.startswith('-Wno-'):
+            if arg.startswith('-Wno-') and '-Wno-attributes=' not in arg:
```

Comment:
Open to bikeshedding on doing this more efficiently, e.g. with compiled regex (lol).

Reply:
Sure.